torchok/tasks/multihead_classification.py

```diff
@@ -0,0 +1,135 @@
+            head_in_channels = self.backbone.out_channels
+        else:
+            pooling_params = pooling_params or dict()
+            self.pooling = POOLINGS.get(pooling_name)(in_channels=pooling_in_channels, **pooling_params)
+            head_in_channels = self.pooling.out_channels
+
+        # HEADS
+        self.heads = nn.ModuleDict()
+        self.target_mapping = {}
+        for head in heads:
+            head_type = HEADS.get(head['type'])
+            head_name = head['name']
+            target = head['target']
+
+            self.heads[head_name] = head_type(in_channels=head_in_channels, **head['params'])
+            self.target_mapping[head_name] = target
+
+    def forward(self, x):
+        features = self.backbone(x)
+        features = self.pooling(features)
+        head_outputs = {}
+        for head_name, head in self.heads.items():
+            head_outputs[head_name] = head(features)
+        return head_outputs
+
+    def forward_with_gt(self, batch):
+        """Forward with ground truth labels.
+
+        Args:
+            batch: Dictionary with the following keys and values:
+
+                - `image` (torch.Tensor):
+                    tensor of shape `(B, C, H, W)`, representing input images.
+                - `target_*` (torch.Tensor):
+                    tensor of shape `(B)`, target class or labels per each image for *-named head.
+                - `condition_*` (Optional[torch.Tensor]):
+                    tensor of shape `(B)`, condition to select appropriate embeddings for the *-named head.
+
+        Returns:
+            Dictionary with the following keys and values
+
+            - 'embeddings': torch.Tensor of shape `(B, num_features)`, representing embeddings per each image.
+            - 'prediction_*':
+                torch.Tensor of shape `(b, num_classes)`, representing logits per each image for *-named head.
+                `b` is less or equal than `B` and depends on `condition_*`.
+            - 'target_*':
+                torch.Tensor of shape `(b)`, target class or labels per each image. May absent.
+                `b` is less or equal than `B` and depends on `condition_*`.
+        """
+
+        features = self.backbone(batch['image'])
+        features = self.pooling(features)
+        output = {'embeddings': features}
+        for head_name, head in self.heads.items():
+            target_name = self.target_mapping[head_name]
+            head_target = batch[f'target_{target_name}']
+            condition = batch.get(f'condition_{target_name}', None)
+
+            if condition is not None:
+                head_target = head_target[condition]
+                out = head(features[condition], head_target)
+            else:
+                out = head(features, head_target)
+
+            output[f'prediction_{head_name}'] = out
+            output[f'target_{target_name}'] = head_target
+
+        return output
+
+    def as_module(self) -> nn.Sequential:
```

Comment:
Why not implemented?)

Reply:
It requires the creation of an ambiguous class for the head. When the exporting to onnx time comes I will think about how to make it